### ISSUE TITLE
upnp / dlna 

### ISSUE THREAD

**Issue by MX10-AC2N** [YunoHost-Apps/jellyfin_ynh#74]
Hi everyone and first a huge thank you for porting Jellyfin under yunohost (special thanks @tituspijean )
I come to you to solve my problem, let me explain.
I am trying to use Jellyfin to play my music on my upnp / dlna Harman Kardon audio amplifier.
First remark I am systematically obliged to retype the command "sudo yunohost firewall allow UDP 1900" each time my server starts up, otherwise jellyfin and my amplifier cannot communicate ..
So after typing this command, my amp sees the Jellyfin server and conversely Jellyfin sees my amp well .. So I can play my music from Jellyfin-web cast on my amp ..
This is where my real problem comes in, at each end of the song I have the beginning of the song which starts over again before finally moving on to the next song, very strange but above all very unpleasant.
 I noticed that the time progression of the songs became Naan: Naan at the end of the song, and I also have some titles which are directly played without a time value which blocks the playback ..
If anyone has an idea to solve this I am interested ..
Hope the solution exists, thanks again ..

**Comment by tituspijean** [comment]
>  I am systematically obliged to retype the command "sudo yunohost firewall allow UDP 1900" each time my server starts up, otherwise jellyfin and my amplifier cannot communicate ..

I will add the opening of the 1900 (server discovery) and 7359 (client discovery) ports in the next update, though it is very weird your server does not remember it after a reboot.

> So after typing this command, my amp sees the Jellyfin server and conversely Jellyfin sees my amp well .. So I can play my music from Jellyfin-web cast on my amp ..
This is where my real problem comes in, at each end of the song I have the beginning of the song which starts over again before finally moving on to the next song, very strange but above all very unpleasant.
I noticed that the time progression of the songs became Naan: Naan at the end of the song, and I also have some titles which are directly played without a time value which blocks the playback ..
If anyone has an idea to solve this I am interested ..
Hope the solution exists, thanks again ..

It feels like an issue on Jellyfin's side, not this package. You should ask for help on their [channels](https://jellyfin.org/contact/) (they have Matric, IRC, and non-free channels). Report back here if it becomes clear I should get involved in fine-tuning the configuration for everybody using this package. 😉

**Comment by MX10-AC2N** [comment]
Thank you for your answers, I will post on the github jellyfin ..
Regarding the opening of the UDP 1900 port I think the yunohost firewall is the cause of the problem.
after restarting my pc, i go to firewall via webadmin and i see that the upnp is disabled, if i try to enable i get an error message. 

Erreur: "500" Internal Server Error

Action: "PUT" /yunohost/api/firewall/upnp/enable

Message d'erreur :
Impossible d’ouvrir les ports UPnP

Pendant le traitement de l'action, le serveur a dit :
Pare-feu rechargé
Le port 1900 est déjà fermé pour les connexions IPv4
Le port 1900 est déjà fermé pour les connexions IPv6
Pare-feu rechargé

It's strange because port 1900 is marked open in TCP IPv4 and IPv6 but is not in the list of UDP ports ??
![Capture d’écran (18)](https://user-images.githubusercontent.com/62643837/118461812-580b0e00-b6fe-11eb-9990-1ed959789464.png)
![Capture d’écran (19)](https://user-images.githubusercontent.com/62643837/118461844-5fcab280-b6fe-11eb-9b8a-5842bbcba9ac.png)
![Capture d’écran (20)](https://user-images.githubusercontent.com/62643837/118461865-65c09380-b6fe-11eb-8dca-8d3bd642b150.png)
So I can't solve anything via Webadmin, unique SSH connection solution and type in CLI "sudo yunohost firewall allow UDP 1900" but if I go back to webadmin to see the open ports I still don't have port 1900 in UDP ??? (but I still find the possible connection on my amp and my TV on Jellyfin)
If I try to open port 1900 UDP under webadmin I have
![Capture d’écran (21)](https://user-images.githubusercontent.com/62643837/118463001-90f7b280-b6ff-11eb-965e-fd01ac2daf1b.png)
In short I think more has a problem in yunohost ...
Hope this helps ..
Thanks @tituspijean

**Comment by tituspijean** [comment]
Ha! That makes me think of a probably similar issue I have with WireGuard. Could you try manually opening the ports for `Both` TCP and UDP?

Regarding the general issue of UPnP with your server, maybe you could ask about it on YunoHost's chat or the forum?

**Comment by MX10-AC2N** [comment]
with Webadmin opening both port 1900 gives
![Capture d’écran (22)](https://user-images.githubusercontent.com/62643837/118466386-192b8700-b703-11eb-8ae4-e61aeed72820.png)
but in CLI under ssh connection the command "sudo yunohost firewall allow BOTH 1900" also allows to find my amp and my tv under Jellyfin
Thank you for your answers, I will open a topic on the yunohost forum ..

**Comment by MX10-AC2N** [comment]
Hi @tituspijean 
I upgraded to branch testing but I did not understand what I need for
" If you are upgrading to this version and above, set discovery: '1' in /etc/yunohost/apps/jellyfin/settings.yml if you want the upgrade to open them for you. "
![Capture d’écran (45)](https://user-images.githubusercontent.com/62643837/120071822-234d6e00-c091-11eb-9492-8dc4bbe9327d.png)
Can you explain to me more clearly what I need to modify in my system.xml file
Thanks again..

**Comment by tituspijean** [comment]
You misread the file path. ;)

Other thing is the discovery setting is only necessary if you need to have the ports opened. This is not your case. ;)

**Comment by tituspijean** [comment]
You misread the file path. ;)

Other thing is the discovery setting is only necessary if you need to have the ports opened. This is not your case. ;)

**Comment by MX10-AC2N** [comment]
Hi and thank you @tituspijean 
I'm too bad, sorry indeed I was not at all in the right place ...
However my problem is still present, when I boot my yunohost server, JF does not see my dlna devices and vice versa, I am still forced to type in CLI "sudo yunohost firewall allow Both 1900"
The problem is probably more at the level of Yunohost..

**Comment by MX10-AC2N** [comment]
Hi @tituspijean, 
This morning I found a new message in my request made on Jellyfin, and I was told to test with the new version of JF
https://github.com/jellyfin/jellyfin/issues/6107#issuecomment-877272303
this refers to a PR of BaronGreenback
https://github.com/jellyfin/jellyfin/pull/4543
So I come back to you because my JF being under YUNoHost I don't really know if I can do anything, apparently a new version 1.8.0 is in progress and DLNA / UPNP is becoming a plugin.
Thank again and again
Nice summer..!

**Comment by tituspijean** [comment]
Hello, thanks for your feedback. Right now there is not much I can do, since the package is already deploying Jellyfin's latest avaible version (10.7.6). I will definitely keep an eye out and make an upgrade available whenever the next version is released. ;)

**Comment by pagiraud** [comment]
DLNA didn’t work for me until last week… Now, my devices recognize each other but I can play media from the server (Jellyfin) to any of my receivers only if I use a controller. Otherwise, the media never starts. Strangely, minidlna is still running and rocking (the documentation argues that Jellyfin can work only if it is the only DLNA server on the domain).

**Comment by Salamandar** [comment]
I can't get devices to discover each other, yet the 1900 and 7359 ports are open in the firewall. Is there something else I can test ? DLNA isn't something I know :/

**Comment by Salamandar** [comment]
I'm pretty sure this is linked to #90.

**Comment by pagiraud** [comment]
Maybe I’m offtopic but it may be linked to your router, may it be the box provided by your ISP or your own. Indeed, I never managed to get the DLNA of Jellyfin to work properly (at best, when I was very lucky, I could browse the directories but nothing played) when I used the box from my previous ISP. Yesterday, I changed my ISP, plugged the new box and… everything works smoothly.

**Comment by tituspijean** [comment]
It is quite possible there is some variations in the implementation of UPNP and DLNA depending on the device or manufacturer.

I personally recently realized that my smart TV had an app store with Jellyfin, and it automatically discovered my server installed on a RPi. No more need to use a casting Android app (that never auto-discovered my server).

**Comment by csolisr** [comment]
Not sure if it's related, but I'm currently having the diagnosis report send me this email every single day:

```
The automatic diagnosis on your YunoHost server identified some issues on your server. You will find a description of the issues below. You can manage those issues in the 'Diagnosis' section in your webadmin.

---

=================================
Ports exposure (ports)
=================================

[ERROR] Port 1900 is not reachable from outside.
  - Exposing this port is needed for [?] features (service jellyfin)
  - To fix this issue, you most probably need to configure port forwarding on your internet router as described in https://yunohost.org/isp_box_config

[ERROR] Port 7359 is not reachable from outside.
  - Exposing this port is needed for [?] features (service jellyfin)
  - To fix this issue, you most probably need to configure port forwarding on your internet router as described in https://yunohost.org/isp_box_config
```

I'm currently using Wireguard to have my personal server access the Internet outside of my network (my ISP does not allow incoming traffic straight to my modem, like my old one did). Do I need to do some special configuration on my Wireguard server? I've already tried routing those two ports directly to my device, but I'm not sure if that's causing more problems instead.

**Comment by tituspijean** [comment]
If the ports are opened on your YunoHost server, that should be OK.

I have just realized these two ports should NOT be publicly accessible. They are for local network discovery (through VPN should be OK too). You are safe to `Ignore` them in the diagnosis, and I will remove the test in a future release of the package.

**Comment by Salamandar** [comment]
> I have just realized these two ports should NOT be publicly accessible. 

Yeah indeed. Had a hard time figuring out why my router didn't want to publicly expose them :D
